### PR TITLE
Add new Mandrel-based Graal native compiling build agent for Java proj

### PR DESCRIPTION
**Pull request by autumnfound**
To build native images for Quarkus, a new agent is needed that uses GraalVM instead of the typical JDK. In testing, Mandrel JDK does not work in Alpine due to the usage of musl, with the compat layer not fixing all of the failures. This image will also include nodejs/yarn/npm for some secondary operations of the builds.

### CODE REVIEW

File: native-build-agent/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# compilation for Quarkus.
+##
+
+ARG MANDREL_VERSION=23.0
+FROM registry.access.redhat.com/quarkus/mandrel-23-rhel8:${MANDREL_VERSION}
+
+ARG NODE_VERSION=v18.20.2
+ARG NPM_VERSION=9.6.6
+ARG YARN_VERSION=1.22.19
+ARG MVN_VERSION=3.8.8
```
Comment (fredg02) [inline native-build-agent/Dockerfile:22]: Is there a good reason to use an old Maven version, instead of the latest one (3.9.6)?

**Review by fredg02 — APPROVED** [review]

**Comment by heurtematte** [comment]
@autumnfound can you add a new stage in the `jenkinsfile`? `Build.sh` script is only for a local build.

**Comment by heurtematte** [comment]
@autumnfound I have noticed that you are 1 commit behind, can you rebase? And add the agent definition to the stage definition.

```groovy
 agent {
            kubernetes {
              yaml loadOverridableResource(
                libraryResource: 'org/eclipsefdn/container/agent.yml'
              )
            }
          }
``

**Comment by autumnfound** [comment]
> @autumnfound I have noticed that you are 1 commit behind, can you rebase? And add the agent definition to the stage definition.

@heurtematte I had just noticed and was in the process of doing that, should be good to go now!

**Comment by autumnfound** [comment]
> error: failed to solve: failed to read dockerfile: failed to create /var/lib/buildkit/runc-native/snapshots/snapshots/1816/.tmp.840971362: open /var/lib/buildkit/runc-native/snapshots/snapshots/1816/.tmp.840971362: no space left on device

Looks like something is up with the storage behind the job :/

**Comment by heurtematte** [comment]
> Looks like something is up with the storage behind the job :/

Yeah, that's my main task this week. I have to find out what's going on there.

**Review by heurtematte — APPROVED** [review]

**Pull request merged**